azure/Guidelines.md

```diff
@@ -0,0 +1,259 @@
+
+The OneAPI guidelines are available on Github at [https://github.com/microsoft/api-guidelines][1].
+
+Highlights of the OneAPI guidelines include:
+
+* **REST fundamentals**.  Guidance around REST basics such as HTTP headers, verbs, status codes, and other similar areas.
+* **Versioning**.  The guidance for versioning covers the externally facing mechanics of versioning and compatibility guarantees between versions.  Guidance for when API owners should increment versions is also present.
+* **CORS**.  Guidance around the use of CORS.
+* **Authentication**.  Guidance around the authentication and its use.
+* **Encoding**.  Guidance around the use of JSON encoding and standards.
+* **Error responses**.  A standard format for error responses is closed.
+* **Dates and times**.  Detailed guidance around date and time formats and encoding is provided.
+* **Relationship to OData**.  Guidance around the relationship between OData and REST APIs.
+* **JSON representation**.  JSON is preferred over XML and other formats.
+* **Long running operations**.  The OneAPI guidelines are updated and simplified over the previous Azure guidelines.
+* **Push notifications via Webhooks**.
+
+#### Asynchronous operations
+
+The OneAPI guidelines for Long Running Operations guidelines are an updated, clarified and simplified version of the Asynchronous Operations guidelines from the 2.1 version of the Azure API guidelines. Unfortunately, to generalize to the whole of Microsoft and not just Azure, the board decided to rename the HEADER used in the operation from `Azure-AsyncOperation` to `Operation-Location`. Services can and **SHOULD** support both `Azure-AsyncOperation` and `Operation-Location` HEADERS, even though they are redundant so that existing SDKs and clients will continue to operate. Clients that call these services **SHOULD** look for both HEADERS and prefer the `Operation-Location` version. Both HEADERS **MUST** return the same value.
+
+### Additional guidance for Azure Resource Manager resource providers
+
+Teams building ARM RPs MUST follow the additional guidance in the ARM RPC and related documents. These documents can be found here.
+
+* [Azure Resource Manager Wiki][2] (Internal only)
+* [Azure Resource Provider Contract][3]
+
+ARM RPs are a CEC requirement for Azure Services and ARM RP review is another mandatory review. Some of the guidance overlaps with general API review, but passing one review will generally make the other one go very quickly.
+
+## Swagger to describe API
+
+All Services **MUST** provide Swagger that describes their service. Swagger is a key element of the Azure SDK plan and essential to improving the usability and discoverability of services. Swagger is a CEC requirement for Azure Services.
```

Comment:
Swagger is a trademarked term.  We should be using OAS terminology.  As we publish the doc we need to snap to correct terms.

Reply:
What is the correct terminology here?

Reply:
I see it is "OpenAPI" - do we have a location where we define the swagger + microsoft definitions (that would be an appropriate link here).  Should we link to the open API hub?

Reply:
Open API Specification (OAS) is the artefact that replaces a swagger doc.  OpenAPI is the org.   @darrelmiller  is best placed to provide the right text here as he's on the board.

Reply:
Changed it to "OpenAPI Specification" per the openapi website.